numpy/core/src/multiarray/descriptor.c

```diff
@@ -519,9 +519,8 @@ _convert_from_array_descr(PyObject *obj, int align)
              || (title
                  && PyBaseString_Check(title)
                  && (PyDict_GetItem(fields, title) != NULL))) {
-            name = PyUnicode_AsUTF8String(name);
             PyErr_Format(PyExc_ValueError,
-                    "field '%s' occurs more than once", PyString_AsString(name));
+                    "field %R occurs more than once", name);
             Py_DECREF(name);
```

Comment:
```suggestion
```
Probably the reason for the test failure.

Reply:
Thanks